### PR TITLE
Change `lines` to lists

### PR DESCRIPTION
**Pull request by DanielNoord**
/CC @cdce8p 

This changes some occurrences of `lines` to become lists

### CODE REVIEW

File: pylint/checkers/base.py [hunk 1]
```diff
@@ -412,22 +412,29 @@ def report_by_type_stats(sect, stats, old_stats):
                 nice_stats[node_type]["percent_badname"] = f"{percent:.2f}"
             except KeyError:
                 nice_stats[node_type]["percent_badname"] = "NC"
-    lines = ("type", "number", "old number", "difference", "%documented", "%badname")
+    lines: list[str] = [
```
Comment (DanielNoord) [inline pylint/checkers/base.py:415]:
```suggestion
    lines: List[str] = [
```

File: pylint/checkers/base.py [hunk 2]
```diff
@@ -412,22 +412,29 @@ def report_by_type_stats(sect, stats, old_stats):
...
                 nice_stats[node_type]["percent_badname"] = "NC"
-    lines = ("type", "number", "old number", "difference", "%documented", "%badname")
+    lines = [
+        "type",
+        "number",
+        "old number",
+        "difference",
+        "%documented",
+        "%badname",
+    ]
```
Comment (Pierre-Sassoulas) [inline pylint/checkers/base.py:422]:
```suggestion
    lines = ["type", "number", "old number", "difference", "%documented", "%badname"]
```

**Review by Pierre-Sassoulas — COMMENTED** [review]
I guess there's a reason for this change, that I don't know of :)

**Review by cdce8p — APPROVED** [review]
LGTM!

Btw. `list[str]` requires `3.9` or `from __future__ import annotations`. `List[str]` would have worked.

**Comment by cdce8p** [comment]
@Pierre-Sassoulas See the discussion here: https://github.com/PyCQA/pylint/pull/4973#discussion_r708572347

**Comment by DanielNoord** [comment]
Yeah I was too quick and then I thought "wait this is implicit"!

**Review by Pierre-Sassoulas — APPROVED** [review]

**Pull request merged**